crates/pallet-grandpa-finality-verifier/src/lib.rs

```diff
@@ -130,41 +137,36 @@ pub mod pallet {
         init_params: InitializationData,
     ) -> DispatchResult {
         let InitializationData {
-            genesis_hash,
+            oldest_parent_height,
+            oldest_parent_hash,
             best_known_finalized_header,
```

Comment:
No, not necessarily same.

I was mainly trying to keep back filling and forward filling possible with also an assumption that validation point is somewhere in future to the known parent.
Example: 1
Validation: 100
OldestParent: 0
0..100..n

Example 2
Validation: 100
OldestParent: 10
10..0
11..100..n

This ensures the pallet supports both stages here - https://github.com/subspace/subspace/issues/379

Reply:
We can ofcourse make them optional and fallback to `best_known_finalized_header `'s parent if that is what you meant

Reply:
Stage 2 replaces stage one. I don't understand why we need gaps of blocks that are not validated at all in between those two, just doesn't make any sense to have older block anything else than `best_known_finalized_header` that we have unconditionally.

This is also to answer the question below about gaps: I assumed you didn't want to upload blocks in between those at all, but now it seems you want to upload them, but not verify (even though you can), so why? :shrug: 

So I don't think they should be optional, they shouldn't be present at all.

Reply:
> Stage 2 replaces stage one.
This is not my assumption so i guess that is where the gap is. Not sure why we had stage 1 though since stage 2 is much simpler in terms of implementation only to be replaced by stage 2 without any use at all.

yes if we were to replace the stage 1 with stage 2, then it will be a much simpler change

Reply:
@rg3l3dr wanted to implement stage 1 first, so :shrug: 

But yeah, we don't need it anymore.